client/pages/plant-detail.jsx

```diff
@@ -1,21 +1,79 @@
 import React from 'react';
+import GardenForm from '../components/garden-form';
 
 export default class PlantDetail extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      plant: null
+      plant: null,
+      btnText: 'Add to garden',
+      gardenCreated: null,
+      modalClass: 'hidden',
+      gardenInfo: {
+        soil: null,
+        sun: null,
+        size: null,
+        notes: ' '
+      }
     };
+    this.handleClick = this.handleClick.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleChange = this.handleChange.bind(this);
   }
 
   componentDidMount() {
     fetch(`http://harvesthelper.herokuapp.com/api/v1/plants/${this.props.plantId}?api_key=${process.env.HARVEST_HELPER_API_KEY}`)
       .then(res => res.json())
       .then(data => {
-        this.setState({
+        return this.setState({
           plant: data
         });
       });
+    fetch('/api/gardenStats')
+      .then(response => response.json())
+      .then(data => {
+        if (data.length !== 0) {
+          this.setState({
+            gardenCreated: true
+          });
+        }
+      });
+  }
+
+  handleClick() {
+    if (!this.state.gardenCreated) {
+      this.setState({
+        modalClass: 'shade'
+      });
+    }
+  }
+
+  handleSave(event) {
+    event.preventDefault();
+    this.setState({
+      modalClass: 'hidden',
+      btnText: 'Remove from garden'
+    });
+    const gardenInfo = this.state.gardenInfo;
+    fetch('/api/gardenStats', {
+      method: 'POST',
+      body: JSON.stringify(gardenInfo),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    })
+      .catch(err => console.error(err));
+  }
+
+  handleChange(event) {
+    const gardenCopy = this.state.gardenInfo;
```

Comment:
To make a copy of an object, you need `Object.assign` with a blank target object

```js
const gardenCopy = Object.assign({}, this.state.gardenInfo);
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign